### DISCUSSION TITLE
How to set action when the done button tapped in PickerInputRow

### DISCUSSION THREAD

**Discussion by repipetokyo** [xmartlabs/Eureka#2157] [Q&A]
I'd like to run a function after the "done" button in pickerView created by PickerInputRow is pushed.

.onCellSelection is called only when the cell is selected, with which you can't get the value after the picker is edited.
If you run a function inside .onChange closure it runs every time the picker value changes, also when the cancel button is tapped, which is not ideal in my case.

Is there any method to achieve this?

**Comment by mats-claassen** [comment]
You could use `.onCellHighlightChanged` which will be called when the focus changes either to or from the picker cell, if that is enough for your case. Note that this will also be called if the user starts to scroll in the form instead of tapping the Done button.

If you need an action to run only when the Done button is tapped then your best chance might be to create a custom NavigationAccessoryView and set it as `customNavigationAccessoryView` of the FormViewController
